examples/gdp/batchProcessing.py

```diff
@@ -0,0 +1,230 @@
+from pyomo.environ import *
+from pyomo.gdp import *
+
+'''Problem from http://www.minlp.org/library/problem/index.php?i=172&lib=GDP
+We are minimizing the cost of a design of a plant with parallel processing units and storage tanks
+in between. We decide the number and volume of units, and the volume and location of the storage
+tanks. The problem is convexified and has a nonlinear objective and global constraints
+
+NOTE: When I refer to 'gams' in the comments, that is Batch101006_BM.gms for now. It's confusing
+because the _opt file is different (It has hard-coded bigM parameters so that each constraint 
+has the "optimal" bigM).'''
+
+model = AbstractModel()
+
+# TODO: it looks like they set a bigM for each j. Which I need to look up how to do...
+model.BigM = Suffix(direction=Suffix.LOCAL)
+model.BigM[None] = 1000
+
+
+## Constants from GAMS
+StorageTankSizeFactor = 2*5 # btw, I know 2*5 is 10... I don't know why it's written this way in GAMS?
+StorageTankSizeFactorByProd = 3
+MinFlow = -log(10000)
+VolumeLB = log(300)
+VolumeUB = log(3500)
+StorageTankSizeLB = log(100)
+StorageTankSizeUB = log(15000)
+UnitsInPhaseUB = log(6)
+UnitsOutOfPhaseUB = log(6)
+# TODO: YOU ARE HERE. YOU HAVEN'T ACTUALLY MADE THESE THE BOUNDS YET, NOR HAVE YOU FIGURED OUT WHOSE
+# BOUNDS THEY ARE. AND THERE ARE MORE IN GAMS.
```

Comment:
Seems unfinished?

Reply:
Yes, that is likely... I ran into some mistakes in the gams file for this problem, and so I know I wasn't ever able to match their results for this one.